src/lib/comms/sol-lwm2m.c

```diff
@@ -163,6 +165,7 @@ struct lifetime_ctx {
 
 struct sol_lwm2m_server {
     struct sol_coap_server *coap;
+    struct sol_coap_transport *transport;
```

Comment:
`sol_coap_server` already keeps a reference to transport, no need to duplicate it here, just add a getter.